--- a/net/minecraft/world/level/block/AmethystClusterBlock.java
+++ b/net/minecraft/world/level/block/AmethystClusterBlock.java
@@ -6,6 +_,7 @@
 import java.util.Map;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
+import net.minecraft.server.level.ServerLevel;
 import net.minecraft.util.RandomSource;
 import net.minecraft.world.item.context.BlockPlaceContext;
 import net.minecraft.world.level.BlockGetter;
@@ -60,6 +_,13 @@
 
     @Override
     protected boolean canSurvive(BlockState state, LevelReader level, BlockPos pos) {
+        // Axium start - per-world update suppression
+        if (level instanceof ServerLevel){
+            if(((ServerLevel) level).paperConfig().updates.suppressBlockUpdates){
+                return true;
+            }
+        }
+        // Axium end
         Direction direction = state.getValue(FACING);
         BlockPos blockPos = pos.relative(direction.getOpposite());
         return level.getBlockState(blockPos).isFaceSturdy(level, blockPos, direction);
