--- a/net/minecraft/world/level/block/FenceBlock.java
+++ b/net/minecraft/world/level/block/FenceBlock.java
@@ -4,6 +_,7 @@
 import java.util.function.Function;
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
+import net.minecraft.server.level.ServerLevel;
 import net.minecraft.tags.BlockTags;
 import net.minecraft.util.RandomSource;
 import net.minecraft.world.InteractionResult;
@@ -104,6 +_,13 @@
         BlockState neighborState,
         RandomSource random
     ) {
+        // Axium start - per-world update suppression
+        if (level instanceof ServerLevel){
+            if(((ServerLevel) level).paperConfig().updates.suppressBlockUpdates){
+                return state;
+            }
+        }
+        // Axium end
         if (state.getValue(WATERLOGGED)) {
             scheduledTickAccess.scheduleTick(pos, Fluids.WATER, Fluids.WATER.getTickDelay(level));
         }
