--- a/net/minecraft/world/level/block/SoulFireBlock.java
+++ b/net/minecraft/world/level/block/SoulFireBlock.java
@@ -38,6 +_,13 @@
 
     @Override
     protected boolean canSurvive(BlockState state, LevelReader level, BlockPos pos) {
+        // Axium start - per-world update suppression
+        if (level instanceof net.minecraft.server.level.ServerLevel){
+            if(((net.minecraft.server.level.ServerLevel) level).paperConfig().updates.suppressBlockUpdates){
+                return true;
+            }
+        }
+        // Axium end
         return canSurviveOnBlock(level.getBlockState(pos.below()));
     }
 
