--- a/net/minecraft/world/level/block/TallSeagrassBlock.java
+++ b/net/minecraft/world/level/block/TallSeagrassBlock.java
@@ -65,6 +_,13 @@
 
     @Override
     protected boolean canSurvive(BlockState state, LevelReader level, BlockPos pos) {
+        // Axium start - per-world update suppression
+        if (level instanceof net.minecraft.server.level.ServerLevel){
+            if(((net.minecraft.server.level.ServerLevel) level).paperConfig().updates.suppressBlockUpdates){
+                return true;
+            }
+        }
+        // Axium end
         if (state.getValue(HALF) == DoubleBlockHalf.UPPER) {
             BlockState blockState = level.getBlockState(pos.below());
             return blockState.is(this) && blockState.getValue(HALF) == DoubleBlockHalf.LOWER;
